pkg/routeagent-driver/handlers/kubeproxy-iptables/endpoint_handler.go

```diff
@@ -0,0 +1,169 @@
+package kp_iptables
+
+import (
+	"fmt"
+	"net"
+
+	"k8s.io/klog"
+
+	submV1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	"github.com/submariner-io/submariner/pkg/util"
+)
+
+func (kp *SyncHandler) LocalEndpointCreated(endpoint *submV1.Endpoint) error {
+	kp.syncHandlerMutex.Lock()
+	defer kp.syncHandlerMutex.Unlock()
+	kp.localCableDriver = endpoint.Spec.Backend
+
+	// We are on nonGateway node
+	if endpoint.Spec.Hostname != kp.hostname {
+		kp.isGatewayNode = false
```

Comment:
May be we should handle this block in TransitionToNonGateway, probably we can pass newEndpoint into the TransitionToNonGateway so this information would become available.

What do you think?

Reply:
Let me think about it...

Reply:
I believe we should do it, let's modify those transition functions to pass down the endpoint info. Otherwise you are replicating the isGateway/Gateway code in here, while it could be done by using the transition functions.

Reply:
> Yeah, I wanted to discuss with you on this.
>
> Naturally if the node was a gateway and transitions to NonGateway, it will be invoked. But what would happen for the following scenarios?


> If admin removes the label from active Gateway node, which translates to deletion of local Endpoint and there will NOT be a new active Gateway node. Will we invoke TransitionToNonGateway in such cases or not? If so, with what endpoint info as argument?

If we need the endpoint to be received we can add that logic into the controller (keeping the previous local endpoint)
 
> The Route-agent comes up (aka bootup) on a nonGateway Node, will TransitionToNonGateway() be called on that node?

Same, ... if you are missing logic that it's necessary on the controller, let's add it.

I think calling TransitionToNonGateway should only be called if we detect that the node is not a gateway at boot up (may be waiting for all endpoints to have synchronized and then sending the transition if we are not gateway..., we could have a timer, or we could improve the admiral watcher to let us know that initial sync finished)